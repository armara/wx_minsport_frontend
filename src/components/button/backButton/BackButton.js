import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'

import Button from 'components/button/Button'

import 'components/button/backButton/BackButton.scss'

const BackButton = ({ text, onClick, history: { goBack } = {} }) => (
  <Button
    className="is-white back-button"
    onClick={onClick || goBack}
    icon="ic-arrow-back"
  >
    {text}
  </Button>
)

BackButton.defaultProps = {
  onClick: null,
}

BackButton.propTypes = {
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  onClick: PropTypes.oneOfType([PropTypes.func, PropTypes.instanceOf(null)]),
  history: PropTypes.shape({
    goBack: PropTypes.func,
  }).isRequired,
}

export default withRouter(BackButton)
