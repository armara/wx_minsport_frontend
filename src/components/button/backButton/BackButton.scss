@import 'styles/mixins';

.back-button {
  @include nav-header-text('16px', '18px', $black);
  margin-top: 5px;
  margin-bottom: 5px;
  height: 60px;
  width: fit-content;

  &.button {
    span.icon:first-child {
      margin-right: 10px;
    }
  }
}