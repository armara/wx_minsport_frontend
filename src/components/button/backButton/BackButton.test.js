import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'

import BackButton from 'components/button/backButton/BackButton'

configure({ adapter: new Adapter() })

describe('<BackButton />', () => {
  const buttonText = 'Вернуться назад'
  const button = mount(<BackButton text={buttonText} />, {
    wrappingComponent: Router,
  })

  it('renders correctly default props', () => {
    const tree = renderer
      .create(
        <Router>
          <BackButton text={buttonText} />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders "text" prop correctly', () => {
    const withStringText = mount(
      <Router>
        <BackButton text={buttonText} />
      </Router>
    )
    const withElementText = mount(
      <Router>
        <BackButton
          text={<div className="back-button-text">{buttonText}</div>}
        />
      </Router>
    )

    expect(withStringText.text()).toEqual(buttonText)
    expect(withElementText.find('.back-button-text').text()).toEqual(buttonText)
  })

  it('is selectable by class ".back-button"', () => {
    expect(button.find('button.back-button').length).toBe(1)
  })

  it('has ".ic-arrow-back" icon', () => {
    expect(button.find('.ic-arrow-back').length).toBe(1)
  })

  it('triggers passed onClick function correctly', () => {
    const mockCallback = jest.fn()
    const buttonWrap = mount(
      <BackButton text={buttonText} onClick={mockCallback} />,
      {
        wrappingComponent: Router,
      }
    )
    buttonWrap
      .find('button.back-button')
      .props()
      .onClick()

    expect(mockCallback.mock.calls.length).toEqual(1)
  })

  it('triggers default goBack function', () => {
    expect(button.find('button.back-button').props().onClick).toBeInstanceOf(
      Function
    )
  })
})
