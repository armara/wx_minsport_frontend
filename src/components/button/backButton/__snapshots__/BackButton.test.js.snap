// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> renders correctly default props 1`] = `
<button
  className="is-white back-button button base-button"
  disabled={false}
  id={null}
  onClick={[Function]}
  title=""
  type="button"
>
  <span
    className="icon"
  >
    <i
      className="ic-arrow-back"
    />
  </span>
  Вернуться назад
</button>
`;
